File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = {};
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i =0; i<players.length; i++){
            map.put(players[i],i);
        }
        //백준허브 테스트
        
        for(int i =0; i<callings.length; i++){
            int pos = map.get(callings[i]);
            String player = players[pos - 1];
            players[pos-1] = callings[i];
            players[pos] = player;
            
            map.replace(player, map.get(player) + 1);
            map.replace(callings[i], map.get(callings[i]) - 1);
        }
        
        //System.out.print(map);
        
        
        
        return players;
    }
    
    
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    answer = []
    player_ranking = {}
    ranking_player = {}
    n = 0
    for p in players:
        player_ranking[p] = n
        ranking_player[n] = p
        n+=1

    for c in callings:
        num = player_ranking[c]
        front_player = ranking_player[num-1]
        
        
        ranking_player[num-1], ranking_player[num] = ranking_player[num], ranking_player[num-1]
        player_ranking[front_player], player_ranking[c] = player_ranking[c], player_ranking[front_player]
        
        

        
    return list(ranking_player.values())